common/chat/conversation.ts

```
@@ -251,12 +251,44 @@ async function removeParticipant(user: User, conversationId: string): Promise<vo
     }
 }
 
+function isStudentChatMember(memberUserId: string): boolean {
+    if (memberUserId.includes('student')) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
 async function markConversationAsReadOnly(conversationId: string): Promise<void> {
     try {
         const conversation = await getConversation(conversationId);
-        const participantIds = Object.keys(conversation.participants);
-        for (const participantId of participantIds) {
-            const response = await fetch(`${talkjsConversationApiUrl}/${conversationId}/participants/${participantId}`, {
+        const memberIds = Object.keys(conversation.participants);
+
+        for (const memberId of memberIds) {
+            const response = await fetch(`${talkjsConversationApiUrl}/${conversationId}/participants/${memberId}`, {
+                method: 'PATCH',
+                headers: {
+                    Authorization: `Bearer ${apiKey}`,
+                    'Content-Type': 'application/json',
+                },
+                body: JSON.stringify({
+                    access: 'Read',
+                }),
+            });
+            await checkResponseStatus(response);
+        }
+    } catch (error) {
+        throw new Error(error);
```

```suggestion
       throw new Error("Could not mark conversation as readonly.", { cause: error })
```

---

Rethrowing is useless unless you want to throw an own error. 🙂